### PR TITLE
Update advanced usage tutorial with tensor observable feature

### PR DESCRIPTION
**Pull request by AroosaIjaz**

### CODE REVIEW

File: beginner/tutorial_advanced_usage.py [hunk 1]
```diff
@@ -56,21 +56,34 @@ def circuit1(param):
...
+@qml.qnode(dev)
+def circuit2(param):
+    qml.RX(param, wires=0)
+    qml.CNOT(wires=[0, 1])
+    return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+print(circuit2(np.pi / 2))
+
+##############################################################################
+# This outcome is expected, as both spins are either up or down.
```
Comment (co9olguy) [inline beginner/tutorial_advanced_usage.py]:
```suggestion
# Notice how this expectation value differs from the local versions above.
```

File: beginner/tutorial_advanced_usage.py [hunk 2]
```diff
@@ -56,21 +56,34 @@ def circuit1(param):
...
 #
-#     It is important to emphasize that the expectation values in ``circuit`` are both **local**,
+#     It is important to emphasize that the expectation values in ``circuit1`` are both **local**,
 #     i.e., this circuit is evaluating :math:`\left\langle \sigma_z\right\rangle_0` and :math:`\left\langle \sigma_z\right\rangle_1`,
-#     not :math:`\left\langle \sigma_z\otimes \sigma_z\right\rangle_{01}` (where the subscript denotes which wires the
-#     observable is located on).
+#     not :math:`\left\langle \sigma_z\otimes \sigma_z\right\rangle_{01}` (where the subscript
+#     denotes which wires the observable is located on). 
+#
+# In order to measure :math:`\langle\sigma_z \otimes \sigma_z \rangle _{01}`, PennyLane's tensor
```
Comment (co9olguy) [inline beginner/tutorial_advanced_usage.py]:
```suggestion
# In order to measure a tensor product observable like :math:`\langle\sigma_z \otimes \sigma_z \rangle _{01}`, 
```

File: beginner/tutorial_advanced_usage.py [hunk 3]
```diff
@@ -56,21 +56,34 @@ def circuit1(param):
...
+# observable feature can be used.
```
Comment (co9olguy) [inline beginner/tutorial_advanced_usage.py]:
```suggestion
# the matrix multiplication operator ``@`` can be used:
```

File: beginner/tutorial_advanced_usage.py [hunk 4]
```diff
@@ -56,21 +56,34 @@ def circuit1(param):
...
+
+##############################################################################
+# This outcome is expected, as both spins are either up or down.
 #
 # We may even mix different return types, for example expectation values and variances:
 
 
 @qml.qnode(dev)
-def circuit1(param):
+def circuit3(param):
```
Comment (co9olguy) [inline beginner/tutorial_advanced_usage.py]: Note that there is already a `circuit3` below, so if we want every circuit to have a different index, we'll need to update below

Comment (josh146) [reply beginner/tutorial_advanced_usage.py]: I generally don't think this is strictly necessary --- most people reading should be familiar with Python, and should follow that a new definition (using the same name) is being created.

Comment (co9olguy) [reply beginner/tutorial_advanced_usage.py]: in that case, we could just call every one `circuit`

File: beginner/tutorial_advanced_usage.py [hunk 5]
```diff
@@ -79,7 +79,7 @@ def circuit2(param):
 
 
 @qml.qnode(dev)
-def circuit3(param):
+def circuit(param):
```
Comment (co9olguy) [inline beginner/tutorial_advanced_usage.py]: @AroosaIjaz note that there is still a `circuit3` below

Comment (AroosaIjaz) [reply beginner/tutorial_advanced_usage.py]: The one in the other section? okay, I will update that too.

**Pull request merged**